File: frontend-offline/src/game/objects/item.ts
import { ANIMS } from "../sprite_animations/animations";
import { MapIndex } from "../types";
import { BaseObj } from "./base";
import type { ItemType } from "../event/events"
export type { ItemType } from "../event/events"
export class ItemObj extends BaseObj {
    itemType: ItemType
    constructor(
        scene: Phaser.Scene,
        index: MapIndex,
        itemType: ItemType
    ) {
        super(scene, index, "items", 0, "item");
        this.itemType = itemType
        this.sprite.anims.play(ANIMS.item[itemType])
    }
}

File: backend/src/game/objects/item.rs
use crate::game::types::{ItemType, MapIndex, ObjType};
use super::base::BaseObj;

pub struct ItemObj {
    pub base: BaseObj,
    pub item_type: ItemType,
}

impl ItemObj {
    pub fn new(index: &MapIndex, item_type: ItemType) -> Self {
        Self {
            base: BaseObj::new(index, ObjType::Item),
            item_type,
        }
    }
}
